node/node.go

```diff
@@ -334,7 +334,7 @@ func createMempoolAndMempoolReactor(config *cfg.Config, proxyApp proxy.AppConns,
 }
 
 func createEvidenceReactor(config *cfg.Config, dbProvider DBProvider,
-	stateDB dbm.DB, blockStore *store.BlockStore, logger log.Logger) (*evidence.Reactor, *evidence.Pool, error) {
+	stateDB dbm.DB, blockStore evidence.BlockStore, logger log.Logger) (*evidence.Reactor, *evidence.Pool, error) {
```

Comment:
We probably shouldn't change this - it's cleaner to use the normal blockstore outside of the evidence module, and only cast it to the `evidence.BlockStore` interface at the boundaries of the `evidence` package. In the future, we may need to access more of the blockstore functionality in this function, which isn't available via the evidence interface.

Reply:
yeah this was actually recommended by the linter

Reply:
Eh, bad linter ;)

Reply:
I suggest we disable the interface linter entirely. It gives lots of false positives, we had issues with this in tm-db and IAVL as well.

Reply:
I'm not too familiar with the interface linter but shall I create a new PR for removing this?

Reply:
lol the interfacer is by Daniel

Reply:
I guess that's cleaner, although I'd be fine with just doing it here as well.

Reply:
and is now archived

Reply:
Yeah it's such a small think I will just do it here I think

Reply:
Haha! He's the one who suggested we disable it in tm-db, because it blocked one of his PR. Yeah, let's disable it.

Reply:
a general question, should these `createX` functions in node be housed in the module and not in node?

Reply:
Nah, they're utility functions for the node.